### PR TITLE
:bug: (kustomize/v2, go/v4): fix issue to don't set kustomizeconfig.yaml when Webhooks are not enable into the project

### LINKED ISSUE #3367: Webhooks manifests has been scaffolded when a project has not webhooks created
### What broke? What's expected?

Currently, the default scaffold will add the patches and the customizartion for webhooks. 
Therefore, ideally would be great we just add the files for webhooks when the tool scaffold webhook.

That means the following code should only be added to the project when the first WebHook is scaffolded

- Update the scaffolds to insert into `config/crd/kustomization.yaml`:

```
# [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix.
# patches here are for enabling the conversion webhook for each CRD
#- patches/webhook_in_installers.yaml
#+kubebuilder:scaffold:crdkustomizewebhookpatch

# the following config is for teaching kustomize how to do kustomization for CRDs.
configurations:
- kustomizeconfig.yaml

```

- Also, add `config/crd/kustomizeconfig.yaml`

### Reproducing this issue

Create a new project and check the webhooks manifests will be scaffolded by default in the patches when no wekbhooks were yet added to the project

### KubeBuilder (CLI) Version

3.10.0

### PROJECT version

3

### Plugin versions

```yaml
go/v4
```

### Other versions

_No response_

### Extra Labels

_No response_

### PR DESCRIPTION
**Pull request by lowang-bh**
## Description

Fix error in the layout when webhooks are not enable in the project. 

## Motivation
Fixes #3367

### CODE REVIEW

File: pkg/plugins/common/kustomize/v2/scaffolds/internal/templates/config/crd/kustomization.go
Comment (camilamacedo86) [inline pkg/plugins/common/kustomize/v2/scaffolds/internal/templates/config/crd/kustomization.go:127]:
```suggestion
# [WEBHOOK] To enable webhook, uncomment the following section
# the following config is for teaching kustomize how to do kustomization for CRDs.

```

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Hi @lowang-bh,

Thank you for your contribution 🥇  
Following some nits and suggestion
- We need to make clear when the user should uncomment therefore I think we need clarify it via comment
- We need to run `make generate` for any change in the scaffold. Could you please run it?
- We need to update the current e2e tests to uncomment this line, see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v4/generate_test.go
- We need to have a new e2e test to test out the solution without webhook so that we ensure that it is not breaking anything. That would be here: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v4/generate_test.go#L69 (a new generate to be called by scaffolding without webhooks)

**Comment by lowang-bh** [comment]
> * We need to run `make generate` for any change in the scaffold. Could you please run it?
> * We need to update the current e2e tests to uncomment this line, see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v4/generate_test.go

Updated.

**Review by camilamacedo86 — COMMENTED** [review]
/ok-to-test

Could you please squash the commits? Also, I think we need to add a test to ensure that the project works when webhooks are not added and those lines are commented? WDYT?

**Comment by lowang-bh** [comment]
> I think we need to add a test to ensure that the project works when webhooks are not added and those lines are commented?

Yes, that's reasonalbe. I am still working on it.

**Comment by Sajiyah-Salat** [comment]
Hello @lowang-bh I would like to help you in this. I understood that webhook should not scaffold on init instead it should be scaffold when we create it. You have made chnages in revelant files. and made a change in test.go. Can I ask on which part you are currently working. Can I help you in this?

**Review by camilamacedo86 — COMMENTED** [review]
@lowang-bh 

It is only missing we add a new test to ensure that the project works fine when no webhooks are enable.

Could you please do that so we can move forward with this one?

PS.: Not that is not a breaking change for users. Who scaffold the project with an previous kubebuilder CLI when upgrade to use its latest version after this PR get merged will not be affected by the changes.

**Comment by lowang-bh** [comment]
> It is only missing we add a new test to ensure that the project works fine when no webhooks are enable

Yes, But I didn't find out how to add a new e2e test for that. Could you please give some advice?

**Comment by lowang-bh** [comment]
> Can I ask on which part you are currently working. Can I help you in this?

Still need a new test to ensure that the project works fine when no webhooks are enable

**Comment by Sajiyah-Salat** [comment]
Hello @camilamacedo86 do we need to remove this part and stop validating webhooks and test out the solution without webhook or we should keep it as is and add new codeblock.
```
By("scaffolding mutating and validating webhooks")
	err = kbc.CreateWebhook(
		"--group", kbc.Group,
		"--version", kbc.Version,
		"--kind", kbc.Kind,
		"--defaulting",
		"--programmatic-validation",
	)
	ExpectWithOffset(1, err).NotTo(HaveOccurred())

	By("implementing the mutating and validating webhooks")
```

**Review by camilamacedo86 — APPROVED** [review]
I think we can move forward here and then work in the e2e test in a follow up PR.
See that I created an issue for we do that: https://github.com/kubernetes-sigs/kubebuilder/issues/3618

Thank you a lot for your contribution to the project. 🥇 

/lgtm

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
The change is falling in the e2e tests:

```
  [FAILED] Unexpected error:
      <*errors.errorString | 0xc000119df0>: 
      make deploy IMG=e2e-test/controller-manager:qcdv failed with error: (exit status 2) /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
      test -s /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin/kustomize || GOBIN=/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin GO111MODULE=on go install sigs.k8s.io/kustomize/kustomize/v5@v5.1.1
      cd config/manager && /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin/kustomize edit set image controller=e2e-test/controller-manager:qcdv
      /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin/kustomize build config/default | kubectl apply -f -
      # Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
      Error: accumulating resources: accumulation err='accumulating resources from '../crd': '/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/config/crd' must resolve to a file': couldn't make target for path '/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/config/crd': invalid Kustomization: yaml: line 21: did not find expected key
      error: no objects passed to apply
      make: *** [Makefile:137: deploy] Error 1
      
      {
          s: "make deploy IMG=e2e-test/controller-manager:qcdv failed with error: (exit status 2) /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin/controller-gen rbac:roleName=manager-role crd webhook paths=\"./...\" output:crd:artifacts:config=config/crd/bases\ntest -s /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin/kustomize || GOBIN=/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin GO111MODULE=on go install sigs.k8s.io/kustomize/kustomize/v5@v5.1.1\ncd config/manager && /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin/kustomize edit set image controller=e2e-test/controller-manager:qcdv\n/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/bin/kustomize build config/default | kubectl apply -f -\n# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.\nError: accumulating resources: accumulation err='accumulating resources from '../crd': '/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/config/crd' must resolve to a file': couldn't make target for path '/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v4/e2e-qcdv/config/crd': invalid Kustomization: yaml: line 21: did not find expected key\nerror: no objects passed to apply\nmake: *** [Makefile:137: deploy] Error 1\n",
      }
  occurred
```

It might be because the changes are no uncomment the the line accordingly, 

To test it out you can for example:
- Have kind or any k8s up and running
- Add a breaking point in the e2e tests so that you can check the scaffold after uncomment the files. See the the temporary dir created with the scaffold to do the tests will be removed always when the tests finishes.

**Review by camilamacedo86 — APPROVED** [review]

**Comment by camilamacedo86** [comment]
/hold 

Until we have the PR for https://github.com/kubernetes-sigs/kubebuilder/issues/3618 so that we can ensure this fix accordingly.

**Comment by camilamacedo86** [comment]
Hi @lowang-bh,

As a follow up we will also need to do like here: https://github.com/kubernetes-sigs/kubebuilder/pull/3629

Could you please check this PR and either give your thoughts.

**Comment by lowang-bh** [comment]
> As a follow up we will also need to do like here: #3629
> 
> Could you please check this PR and either give your thoughts.

What shall I do now?

**Comment by lowang-bh** [comment]
@camilamacedo86 Is this pr still needed or not since #3629 has been merged?

File: docs/book/src/component-config-tutorial/testdata/project/config/crd/kustomization.yaml
Comment (camilamacedo86) [inline docs/book/src/component-config-tutorial/testdata/project/config/crd/kustomization.yaml:23]:
```suggestion
#configurations:
#- kustomizeconfig.yaml
```
We cannot have the space 
Also, we should automatically uncomment this one when a webhook is scaffold.

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Hi @lowang-bh,

By doing the e2e tests I could check that it would still missing few things 
So, I used your PR to give your the credit as well and keep you as co-author 
See; https://github.com/kubernetes-sigs/kubebuilder/pull/3647


Can we close this one in favor of https://github.com/kubernetes-sigs/kubebuilder/pull/3647
Could you please help in the review of https://github.com/kubernetes-sigs/kubebuilder/pull/3647 for we move forward and get that one merged.  

/approved cancel
/lgtm cancel

**Pull request closed without merging**
